metadata-ws/src/test/resources/features/analysis.feature

```diff
@@ -0,0 +1,394 @@
+Feature: analysis object
+
+  Scenario: register an analysis successfully
+    When I request POST /reference-sequences with JSON payload:
+    """
+    {
+      "name": "GRCh37",
+      "patch": "p2",
+      "accessions": ["GCA_000001407.3", "GCF_000001407.14"],
+      "type": "ASSEMBLY"
+    }
+    """
+    Then set the URL to TEST_REFERENCE_SEQUENCE
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9606,
+      "name": "Homo Sapiens"
+    }
+    """
+    Then set the URL to TEST_TAXONOMY
+    When I create a test study with TEST_TAXONOMY for taxonomy
+    Then set the URL to TEST_STUDY
+    When I create a test analysis with TEST_STUDY for study and TEST_REFERENCE_SEQUENCE for reference sequence
+    Then the response code should be 201
+    And set the URL to TEST_ANALYSIS
+    When I request GET with value of TEST_ANALYSIS
+    Then the response code should be 200
+    And the result should have accessionVersionId.accession with value EGAA0001
+
+  Scenario: update an analysis successfully
+    When I request POST /reference-sequences with JSON payload:
+    """
+    {
+      "name": "GRCh37",
+      "patch": "p2",
+      "accessions": ["GCA_000001407.3", "GCF_000001407.14"],
+      "type": "ASSEMBLY"
+    }
+    """
+    Then set the URL to TEST_REFERENCE_SEQUENCE_1
+    When I request POST /taxonomies with JSON payload:
+    """
+    {
+      "taxonomyId": 9606,
+      "name": "Homo Sapiens"
+    }
+    """
+    Then set the URL to TEST_TAXONOMY
+    When I create a test study with TEST_TAXONOMY for taxonomy
+    Then set the URL to TEST_STUDY
+
+    When I create a test analysis with TEST_STUDY for study and TEST_REFERENCE_SEQUENCE_1 for reference sequence
+    Then the response code should be 201
+    And set the URL to TEST_ANALYSIS
+    When I request GET with value of TEST_ANALYSIS
+    Then the response code should be 200
+    And the result should have accessionVersionId.accession with value EGAA0001
+    When I request GET for referenceSequences of TEST_ANALYSIS
+    Then the response code should be 200
+    And the result should contain 1 reference-sequences
```

Comment:
"contain 1 reference-sequence**s** " is odd

Reply:
`reference-sequences` is class name in json body.

Reply:
ya you can rephrase the line suitable for singular and plural cases

Reply:
ok, "And the result should contain 1 `reference-sequence` " will be handled separately in reference-sequence specific class